### DISCUSSION TITLE
How to use Early Stopping callback with DeepVAR?

### DISCUSSION THREAD

**Discussion by mtomic123** [awslabs/gluonts#2657] [Q&A]
This page: `https://ts.gluon.ai/stable/tutorials/advanced_topics/trainer_callbacks.html `outlines how to use early stopping with a model and it provides a Class `MetricInferenceEarlyStopping` for that.

How does one use this for my case where I define the predictor at the end? For example in my case I have:

```
# Callback for early stopping
es_callback = MetricInferenceEarlyStopping(
    validation_dataset=df_test, predictor=predictor, metric="MSE")


estimator = DeepVAREstimator(
    freq="1H",
    prediction_length=prediction_length,
    trainer=Trainer(epochs=10, callbacks=[history, es_callback]),
    target_dim=10,
    num_layers=num_layers,
    num_cells=num_cells,
    context_length=context_length,)

predictor = estimator.train(training_data, df_test)
```

however the issue here is that `es_callback` requires predictor before I create a predictor?

**Comment by ShirleyMgit** [comment]
@mtomic123 is it running without this callback? how did you introduce that test set?
